contracts/TokenManager.sol

```diff
@@ -221,16 +258,32 @@ contract TokenManager is ITokenController, IForwarder, AragonApp {
     * @return False if the controller does not authorize the transfer
     */
     function onTransfer(address _from, address _to, uint256 _amount) external onlyToken returns (bool) {
-        return _isBalanceIncreaseAllowed(_to, _amount) && _transferableBalance(_from, getTimestamp()) >= _amount;
+        bool transferable = _isBalanceIncreaseAllowed(_to, _amount) && _transferableBalance(_from, getTimestamp()) >= _amount;
+        uint256 i = 0;
+        while (transferable && i < hooksLength) {
+            if (address(hooks[i]) != 0) {
+                transferable = transferable && hooks[i].onTransfer(_from, _to, _amount);
```

Comment:
Isn't `transferable` always `true` here and therefore can be removed from the right side of the setter?

Reply:
Please take into account that it is an AND, not an OR, so `transferable` is false as soon one of the hooks returns false.

Reply:
But surely as soon as `transferable` is false the while statement fails and this assignment statement is never reached?